recipes/finalize.rb

```diff
@@ -15,14 +15,29 @@
 # OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Restart supervisord
-service "supervisord" do
-  supports restart: true
-  action %i[enable start]
+case node['cfncluster']['cfn_scheduler']
+when 'slurm'
+  # disable supervisord when scheduler is slurm
+  service "supervisord" do
+    action %i[disable stop]
+  end
+else
+  # Restart supervisord for other schedulers
+  service "supervisord" do
+    supports restart: true
+    action %i[enable start]
+  end
 end
 
-execute "compute_ready" do
-  command "/opt/parallelcluster/scripts/compute_ready"
-  environment('PATH' => '/usr/local/bin:/usr/bin/:$PATH')
-  only_if { node['cfncluster']['cfn_node_type'] == 'ComputeFleet' }
+return unless node['cfncluster']['cfn_node_type'] == 'ComputeFleet'
+
+case node['cfncluster']['cfn_scheduler']
+when 'slurm'
+  include_recipe 'aws-parallelcluster::_compute_slurm_finalize'
+else
+  execute "compute_ready" do
+    command "/opt/parallelcluster/scripts/compute_ready"
+    environment('PATH' => '/usr/local/bin:/usr/bin/:$PATH')
+    only_if { node['cfncluster']['cfn_node_type'] == 'ComputeFleet' }
```

Comment:
this is a duplicated condition. We could simplify all of this by removing the return and the case and moving this in the previous case on scheduler by simply adding an if on the node type.